File: test-suite/src/test/java/io/micronaut/docs/qualifiers/annotationmember/V6Engine.java
package io.micronaut.docs.qualifiers.annotationmember;

import jakarta.inject.Singleton;

// tag::class[]
@Singleton
@Cylinders(value = 6, description = "6-cylinder V6 engine")  // <1>
public class V6Engine implements Engine { // <2>

    @Override
    public int getCylinders() {
        return 6;
    }

    @Override
    public String start() {
        return "Starting V6";
    }
}
// end::class[]

File: test-suite-kotlin-ksp/src/test/kotlin/io/micronaut/docs/qualifiers/annotationmember/V6Engine.kt
package io.micronaut.docs.qualifiers.annotationmember

import jakarta.inject.Singleton

// tag::class[]
@Singleton
@Cylinders(value = 6, description = "6-cylinder V6 engine") // <1>
class V6Engine : Engine { // <2>
    // <2>
    override val cylinders: Int
        get() = 6

    override fun start(): String {
        return "Starting V6"
    }
}
// end::class[]
